pkg/agent/controller/traceflow/packetin.go

```diff
@@ -100,6 +101,26 @@ func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*opsv1alpha1.Tracefl
 		obs = append(obs, *ob)
 	}
 
+	// Collect service DNAT.
+	if pktIn.Data.Ethertype == 0x800 {
+		ipPacket, ok := pktIn.Data.Data.(*protocol.IPv4)
+		if !ok {
+			return nil, nil, errors.New("invalid traceflow IPv4 packet")
+		}
+		ctNwDst, err := getInfoInCtNwDstField(matchers)
+		if err != nil {
+			return nil, nil, err
+		}
+		ipDst := ipPacket.NWDst.String()
+		if ctNwDst != "" && ipDst != ctNwDst {
+			ob := new(opsv1alpha1.Observation)
+			ob.Component = opsv1alpha1.LB
+			ob.Action = opsv1alpha1.Forwarded
+			ob.TranslatedDstIP = ipDst
```

Comment:
I think we tend to use a different style in the rest of the Antrea code base:
```
ob := &opsv1alpha1.Observation{Component: ...}
```

Reply:
Changed to
```
			ob := &opsv1alpha1.Observation{
				Component:       opsv1alpha1.LB,
				Action:          opsv1alpha1.Forwarded,
				TranslatedDstIP: ipDst,
			}
```